addons/sale/sale.py

```diff
@@ -661,6 +661,8 @@ def write(self, values):
     product_id = fields.Many2one('product.product', string='Product', domain=[('sale_ok', '=', True)], change_default=True, ondelete='restrict', required=True)
     product_uom_qty = fields.Float(string='Quantity', digits=dp.get_precision('Product Unit of Measure'), required=True, default=1.0)
     product_uom = fields.Many2one('product.uom', string='Unit of Measure', required=True)
+    #We've added a related field to store the category of the UoM for this line. We use this as a default domain in the view.
+    relcatid = fields.Many2one(related='product_uom.category_id', store=True)
```

Comment:
not compatible with [stable policy](https://github.com/odoo/odoo/wiki/Contributing#what-does-stable-mean)